packages/process/src/node/process-manager.ts

```diff
@@ -63,13 +66,32 @@ export class ProcessManager implements BackendApplicationContribution {
         }
         if (this.processes.delete(process.id)) {
             this.deleteEmitter.fire(process.id);
+            // Process will be recorded in willReapProcesses for a while so that it's still possible to get information of it before it's automatically reaped
+            this.willReapProcesses.set(process.id, process);
+            setTimeout(() => {
+                this.willReapProcesses.delete(process.id);
+            }, this.REAP_TIMEOUT);
```

Comment:
I did exactly the same thing in my local environment to get around this bug.
I am not sure if this is how it should be fixed. I will leave it to @marechal-p  and @akosyakov :)

Reply:
@elaihau Haha, what a coincidence. The idea is that if one wants to manipulate, for example, write to that process, or resize the terminal process and so on, the process should be alive, and `getAlive` may be called (a more reliable way is to get the process by either method, and check its `killed` property right before manipulating it); otherwise, `get` method can be used and, although the process may be dead, the caller can still get the output stream or other attached information to that process. The `getAlive` method is just equivalent to the former `get` method.

Reply:
I think the proper solution is to have clear lifecycles and don't rely on timings: https://github.com/eclipse-theia/theia/issues/2961#issuecomment-626591318